app/Domain/Goalcanvas/Services/Goalcanvas.php

```diff
@@ -202,5 +223,488 @@ public function pollForUpdatedGoals(): array|false
+            } elseif (session()->exists($sessionKey)) {
+                $currentCanvasId = session($sessionKey);
+                $currentCanvasId = $this->validateCanvasId($currentCanvasId, $allCanvas);
+            } else {
+                $currentCanvasId = $allCanvas[0]['id'] ?? -1;
+            }
+
+            session([$sessionKey => $currentCanvasId]);
+
+            return $currentCanvasId;
+        }
+
+        private function validateCanvasId(int $canvasId, array $allCanvas): int
+        {
+            foreach ($allCanvas as $canvas) {
+                if ($canvasId == $canvas['id']) {
+                    return $canvasId;
+                }
+            }
+            return -1;
+        }
+
+        // Goal Dashboard Post
+
+        public function handleDashboardPostRequest($params): array
+        {
+            $result = [
+                'redirect' => false,
+                'redirectUrl' => '',
+                'notification' => null,
+            ];
+
+            $currentCanvasId = session("current" . strtoupper(static::CANVAS_NAME) . "Canvas") ?? -1;
+
+            if (isset($params['newCanvas'])) {
+                $result = $this->handleNewCanvas($params, $result);
+            } elseif (isset($params['editCanvas']) && $currentCanvasId > 0) {
+                $result = $this->handleEditCanvas($params, $currentCanvasId, $result);
+            } elseif (isset($params['cloneCanvas']) && $currentCanvasId > 0) {
+                $result = $this->handleCloneCanvas($params, $currentCanvasId, $result);
+            } elseif (isset($params['mergeCanvas']) && $currentCanvasId > 0) {
+                $result = $this->handleMergeCanvas($params, $currentCanvasId, $result);
+            } elseif (isset($params['importCanvas'])) {
+                $result = $this->handleImportCanvas($params, $result);
+            }
+
+            return $result;
+        }
+
+        private function handleNewCanvas($params, $result): array
+        {
+            if (!isset($params['canvastitle']) || empty($params['canvastitle'])) {
+                return $this->setNotification($result, 'notification.please_enter_title', 'error');
+            }
+
+            if ($this->goalRepository->existCanvas(session("currentProject"), $params['canvastitle'])) {
+                return $this->setNotification($result, 'notification.board_exists', 'error');
+            }
+
+            $values = [
+                'title' => $params['canvastitle'],
+                'author' => session("userdata.id"),
+                'projectId' => session("currentProject"),
+            ];
+            $currentCanvasId = $this->goalRepository->addCanvas($values);
+
+            $this->sendCanvasNotification('notification.board_created', $values['title']);
+
+            session(["current" . strtoupper(static::CANVAS_NAME) . "Canvas" => $currentCanvasId]);
+            return $this->setRedirect($result, BASE_URL . '/' . static::CANVAS_NAME . 'canvas/showCanvas/');
```

Comment:
The service layer should ideally not perform any redirects. If the service is called via API the redirect would break the api response. 

I think in this case the method could just return either an error message or the id of the new canvas.